controllers/slice/reconciler.go

```diff
@@ -156,17 +156,17 @@ func (r *SliceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl
 		return ctrl.Result{}, err
 	}
 
-	if slice.Status.SliceConfig.SliceOverlayNetworkDeploymentMode == v1alpha1.NONET {
-		log.Info("No communication slice, skipping reconciliation of appns, qos, netop, egw, router etc.")
-		return ctrl.Result{}, nil
-	}
-
 	res, err, requeue := r.ReconcileSliceNamespaces(ctx, slice)
 	if requeue {
 		debugLog.Info("Reconciling SliceNamespaces", "res", res, "err", err)
 		return res, err
 	}
 
+	if slice.Status.SliceConfig.SliceOverlayNetworkDeploymentMode == v1alpha1.NONET {
```

Comment:
I think we need to move all the networking related statements into a new func. 

func ReconcileNetworkingFeatures() {
    reconcile_slice_router
    reconcile_slicegw_edge
    reconcile_ext_gateways
    syncQoStoNetop
}

func Check_and_delete_networking_features() {
    delete_slice_router_if_present
    delete_slicegw_edge_if_present
    delete_ext_gateways_if_present
}


if single-network || multi-network {
     ReconcileNetworkingFeatures()
} else {
     Check_and_delete_networking_features()
}

Reconcile_App_pods()


We would still need to reconcile app pods that are part of the slice. There is an app pod reconcile func below.

Reply:
Addressed. Ptal